lua/nvim-treesitter/ts_utils.lua

```diff
@@ -0,0 +1,103 @@
+local api = vim.api
+
+local M = {}
+
+--- Gets the actual text content of a node
+-- @param node the node to get the text from
+-- @param bufnr the buffer containing the node
+-- @return list of lines of text of the node
+function M.get_node_text(node, bufnr)
+  local bufnr = bufnr or api.nvim_get_current_buf()
+  if not node then return {} end
+
+  -- We have to remember that end_col is end-exclusive
+  local start_row, start_col, end_row, end_col = node:range()
```

Comment:
I really hate `node:range()`. We could think about using the LSP ranges in another refactor PR:
```lua
function M.node_to_lsp_range(node)
  local start_line, start_col, end_line, end_col = node:range()
  local rtn = {}
  rtn.start = { line = start_line, character = start_col }
  rtn['end'] = { line = end_line, character = end_col }
  return rtn
end
```

Reply:
i agree it is quite disturbing, we should do something like that indeed.

Reply:
Or we should modify the way `node:range` works on neovim's side.